tests/services/test_ingestion.py

```diff
@@ -118,14 +126,30 @@ def test_extract_schema_and_xlsx_failures(app, url, data, error, message):
             extract_schema_and_xlsx()
 
 
-def test_upload_manifest(app_no_auth, pbmc_valid_xlsx):
+def test_upload_manifest_non_existing_trial_id(
+    app_no_auth, pbmc_non_existing_trial, test_user, db_with_trial_and_user
+):
+    """Ensure the upload_manifest endpoint follows the expected execution flow"""
+    client = app_no_auth.test_client()
+
+    res = client.post(
+        MANIFEST_UPLOAD, data=form_data("pbmc.xlsx", pbmc_non_existing_trial, "pbmc")
+    )
+    assert res.status_code == 400
+    assert "test-non-existing-trial-id" in res.json["_error"]["message"]
+
+
+def test_upload_manifest(
+    app_no_auth, pbmc_valid_xlsx, test_user, db_with_trial_and_user
+):
     """Ensure the upload_manifest endpoint follows the expected execution flow"""
     client = app_no_auth.test_client()
 
     res = client.post(
         MANIFEST_UPLOAD, data=form_data("pbmc.xlsx", pbmc_valid_xlsx, "pbmc")
     )
-    assert res.status_code == 501  # Not Implemented
+    res_body_json = res.json
```

Comment:
Is `res_body_json` used? Also, `# Not yet Implemented` no longer applies, right?